crowdfunding/projects/models.py

```diff
@@ -1,3 +1,90 @@
 from django.db import models
+from django.contrib.auth import get_user_model
 
 # Create your models here.
+
+# for every model, create a serializer
+
+User = get_user_model() # added to use the user
+class Project(models.Model):
+    title = models.CharField(max_length=100, unique=True)
+    description = models.TextField()
+    goal = models.IntegerField() #want to set a min of 1
+    image = models.URLField()
+    is_open = models.BooleanField() #alt = is_active or status
+    date_created = models.DateTimeField(auto_now_add=True) #TIP: auto_now_add=True... will update to time when created
+    owner = models.ForeignKey( # the below connects the ID of the owner to the owner_projects
+        User,
+        on_delete=models.CASCADE,
+        related_name='owner_projects'
+        )
+
+    @property
+    def sum_pledges(self):
+        '''
+        Calculates the total of each pledge for each project.
+        '''
+        pledge_sum = self.pledges.aggregate(sum=models.Sum('amount'))['sum']
+        if pledge_sum == None:
+            return 0
+        else:
+            return pledge_sum
+
+    @property
+    def goal_balance(self):
+        return self.goal - self.sum_pledges
+
+    @property
+    def funding_status(self):
+        if self.goal_balance <= 0:
+            return f"Fully funded"
+        elif self.goal_balance == self.goal:
+            return f"No pledges received"
+        else:
+            return f"Partially funded"
```

Comment:
love all these properties! 
however i would probably caution against returning a string from the backend.
The back-end should be impartial, it's all about the logic and the data. 
it should look at this as being a "True, False, None" situation, or perhaps a set of codes like `("full", "partial", "none") which could be interpreted by the font end.
it's the job of the front end to translate the computer interpreted data to human information.
refuse the temptation to interpret your data young padawan